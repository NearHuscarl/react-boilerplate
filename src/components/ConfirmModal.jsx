import React from 'react';
import Modal from 'react-modal';

const ConfirmModal = (props) => (
	<Modal
		isOpen={props.isOpen}
		onRequestClose={props.onRequestClose}
		contentLabel="Confirm Delete"
		closeTimeoutMS={200}
		className="modal"
		ariaHideApp={false}>
		<h3 className="modal__title">{props.prompt}</h3>
		<button className="button modal__button" onClick={props.onConfirm}>
			yes
		</button>
		<button className="button modal__button" onClick={props.onCancel}>
			no
		</button>
	</Modal>
);

export default ConfirmModal;
