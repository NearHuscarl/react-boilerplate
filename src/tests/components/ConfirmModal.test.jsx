import React from 'react';
import { shallow } from 'enzyme';
import ConfirmModal from '../../components/ConfirmModal';

test('should render ConfirmModal correctly', () => {
	const wrapper = shallow(<ConfirmModal />);
	expect(wrapper).toMatchSnapshot();
});
