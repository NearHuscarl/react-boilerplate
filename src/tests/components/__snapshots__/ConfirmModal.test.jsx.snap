// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConfirmModal correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Confirm Delete"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal__title"
  />
  <button
    className="button modal__button"
  >
    yes
  </button>
  <button
    className="button modal__button"
  >
    no
  </button>
</Modal>
`;
